src/components/ButtonWithLoading/index.js

```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import { Button } from "../Button";
+import Loader from "./Loader";
+
+const ButtonWithLoading = ({ isLoading, children, ...props }) => {
+  const { variant } = props;
+
+  return (
+    <Button {...props}>
+      {isLoading ? children : <Loader variant={variant}>{children}</Loader>}
```

Comment:
in order to make sure we don't put block elements inside `span` elements which are in `Loader` component we shouldn't pass the children to `Loader`

Reply:
You need the children to expand the button element I guess. In this case it seems that we shouldn't do conditional rendering rather than add visibility hidden to children when in loading state and remove `ContentWrapper` from `Loader`

Reply:
Discussed.